packages/cli/commands/project/dev.js

```diff
@@ -93,6 +93,9 @@ exports.handler = async options => {
   if (hasPrivateApps && hasPublicApps) {
     logger.error(i18n(`${i18nKey}.errors.invalidProjectComponents`));
     process.exit(EXIT_CODES.SUCCESS);
+  } else if (!hasPrivateApps && !hasPublicApps) {
+    logger.error(i18n(`${i18nKey}.errors.noSupportedComponents`));
+    process.exit(EXIT_CODES.SUCCESS);
```

Comment:
Should this be `EXIT_CODES.ERROR`?

Reply:
Also, I just saw the `if` statement above it is also a`EXIT_CODES.SUCCESS`, but it feels like that should also be an error

Reply:
Not 100% sure if this is correct, but I've been using `SUCCESS` to indicate that the command functioned as intended. I.e. an invalid project state doesn't indicate any problems with how the dev command actually ran, so its still successful

Reply:
Interesting. Do we have this recorded in our guidelines anywhere? I've been following a similar pattern to what Camden mentioned. This seems like something we should have clear guidelines for though. Technically the command succeeded, but the project that you ran the command on wasn't configured in a way that let the command do everything it wanted to do. 

There's a breakdown of common exit codes [here](https://tldp.org/LDP/abs/html/exitcodes.html). None of them seem to capture this case, which makes me think that success is the right move here.

Reply:
Interesting, I've always thought of it as the `SUCCESS` code means that it made it all the way through the happy path, so I looked at what a couple of common cli utils do and they contradict

```
🚀🚀🚀 ➜ cat notReal; echo $?  
cat: notReal: No such file or directory
0

🚀🚀🚀 ➜ ls notReal; echo $?
ls: notReal: No such file or directory
1

```


I think this is something we might want to standardize in the long term, but that's no something we need to worry about in this PR!

I'll mark this resolved and ✅

Reply:
Yeah, it's not consistent at all within the CLI either. From the little research I've done into exit codes they seem fairly inconsequential (and i guess `cat` and `ls` working differently is evidence for this), but totally agree that we should decide on one approach and use it consistently